src/main/java/org/breedinginsight/brapps/importer/services/processors/ProcessorManager.java

```diff
@@ -0,0 +1,112 @@
+ * limitations under the License.
+ */
+package org.breedinginsight.brapps.importer.services.processors;
+
+import io.micronaut.context.annotation.Prototype;
+import lombok.extern.slf4j.Slf4j;
+import org.breedinginsight.brapps.importer.model.ImportUpload;
+import org.breedinginsight.brapps.importer.model.imports.BrAPIImport;
+import org.breedinginsight.brapps.importer.model.imports.PendingImport;
+import org.breedinginsight.brapps.importer.model.response.ImportPreviewResponse;
+import org.breedinginsight.brapps.importer.model.response.ImportPreviewStatistics;
+import org.breedinginsight.brapps.importer.services.ImportStatusService;
+import org.breedinginsight.model.Program;
+import org.breedinginsight.services.exceptions.ValidatorException;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Prototype
+@Slf4j
+public class ProcessorManager {
+
+    private List<Processor> processors;
+    private Map<Integer, PendingImport> mappedBrAPIImport;
+    private ImportStatusService statusService;
+    private Map<String, ImportPreviewStatistics> statistics = new HashMap<>();
+
+    @Inject
+    public ProcessorManager(ImportStatusService statusService) {
+        this.processors = new ArrayList<>();
+        this.mappedBrAPIImport = new HashMap<>();
+        this.statusService = statusService;
+    }
+
+    public ImportPreviewResponse process(List<BrAPIImport> importRows, List<Processor> processors, Program program, ImportUpload upload, boolean commit) throws ValidatorException {
+
+        this.processors = processors;
+        statusService.setUpload(upload);
+
+        // check existing brapi objects and map data for each registered type
+        for (Processor processor : processors) {
+            statusService.updateMessage("Checking existing " + processor.getName().toLowerCase() + " objects in brapi service and mapping data");
+            Map<String, ImportPreviewStatistics> stats = processor.process(importRows, mappedBrAPIImport, program);
+            statistics.putAll(stats);
+            //processor.getExistingBrapiObjects(importRows, program);
+        }
+
+        ImportPreviewResponse response = new ImportPreviewResponse();
+        response.setStatistics(statistics);
+        List<PendingImport> mappedBrAPIImportList = new ArrayList<>(mappedBrAPIImport.values());
+        response.setRows(mappedBrAPIImportList);
+
+        statusService.updateMappedData(response, "Finished mapping data to brapi objects");
+
+        if (!commit) {
+            statusService.updateOk();
+            return response;
+        } else {
+            validateProcessorDependencies();
+            postBrapiData(program);
+        }
+
+        return response;
+    }
+
+    private void validateProcessorDependencies() {
+        for (Processor processor : processors) {
+            try {
+                processor.validateDependencies(mappedBrAPIImport);
```

Comment:
Ya the intention of these is to check all data dependencies prior to posting brapi data so that we can catch and return errors we know about before we kick off all of the processors running their posts. This would be part of an error checking effort in the importer. Not sure if there's already a card of if there should be a separate one specifically for this?

Reply:
Oh yeah, that way we don't POST half of the stuff and then run into a dependency error that prevents the rest of the creation, good move! I don't remember there being a card, we should probably add one for this though so we don't forget.

Reply:
Created card: https://breedinginsight.atlassian.net/browse/BI-1003?atlOrigin=eyJpIjoiZjc4ODM3ZDRkNTNjNDc4Y2I1ZTVjMDRjZThkOWRmZTkiLCJwIjoiaiJ9